### PR TITLE
Add Scripts for Turning a RPI0 into a HID

### PR DESCRIPTION
**Pull request by erichlf**
This is what we need to get the rpi zero to be a human interface device. There is still plenty of work to get this to intercept key presses on your keyboard though.

### CODE REVIEW

**Comment by jamesclampffer** [comment]
Will get to this in the next couple days.  Pretty sure I have an old ARM dev board with USB OTG sitting around, if I can't find it today I'll put in an order for one that does.

File: hid/testHID.py
Comment (jamesclampffer) [inline hid/testHID.py:15]: I think you could simplify this by doing something like:
```
for char in "Hemiola\nworks":
    write_report( NULL_CHAR*2 + char + NULL_CHAR*5 )
```
Unless this is also intended to be an example of how to press keys etc.

Comment (erichlf) [reply hid/testHID.py:15]: It is an example of how key presses work, but also I am not printing the `char` directly. Those are the escape codes. Also, notice that the lower case key codes come first, which is not how it works in ascii.
You can checkout the inspiration [here](https://randomnerdtutorials.com/raspberry-pi-zero-usb-keyboard-hid/).

Comment (jamesclampffer) [reply hid/testHID.py:15]: Ah, I didn't realize they didn't directly convert to ascii.  I still think it could be compacted to a lambda/function but either way makes it nice example of how to write some bytes out to the host as a HID.

Code looks good but I'm waiting on similar hardware to test it myself. If this is a blocker, or frankly a moderate inconvenience, for future development don't wait on me.  I'll have upcoming time to allocate to actually writing some code.

Marking resolved: not able to test it myself but don't want to bottleneck.  We can always test again once I have proper hardware and merge a subsequent patch to make any fixes.

Semi-related:
If there is a direct mapping from `{key->byte}` it might make life easier to just make a 256 byte lookup table from `{keyboard_key->ascii_byte}` at some point.  Then arbitrary keyboard remappings can be done with no performance overhead.

**Review by jamesclampffer — APPROVED** [review]

File: hid/testHID.py
Comment (erichlf) [reply hid/testHID.py:15]: This python script is not very important. It is purely for people to verify they set things up correctly.

**Comment by erichlf** [comment]
I'm going to merge this and if there are issues we'll address them as they come up.

**Pull request merged**
